zerver/lib/github.py

```diff
@@ -61,3 +66,42 @@ def get_latest_github_release_download_link_for_platform(platform: str) -> str:
         if verify_release_download_link(link):
             return link
     return "https://github.com/zulip/zulip-desktop/releases/latest"
+
+
+def get_issue_or_pr_cache_key(owner: str, repo: str, issue_number: str) -> str:
+    return f"github_issue_or_pr_data:{owner}/{repo}/{issue_number}"
+
+
+@cache_with_key(get_issue_or_pr_cache_key, timeout=3600)
+def get_issue_or_pr_data(owner: str, repo: str, issue_number: str) -> Dict[str, str]:
+    api_url = f"https://api.github.com/repos/{owner}/{repo}/issues/{issue_number}"
+    try:
+        response = GithubSession().get(api_url)
+        response.raise_for_status()
+        data: Dict[str, Any] = response.json()
+        additional_data = {
+            "result": "success",
+            "platform": "github",
+            "type": "pulls" if "pull_request" in data else "issues",
+            "owner": owner,
+            "repo": repo,
+            "issue_number": issue_number,
+        }
+        data.update(additional_data)
+        return data
+    except requests.RequestException as err:
+        logger.exception("Unable to fetch data from GitHub %s", api_url, stack_info=True)
```

Comment:
I think we should be a little more careful with the log level here, based on the status code:
 - 5xx error codes: there's nothing the server administrator can do, so filling out the Zulip server error logs isn't going to help -- we should use INFO at most.
 - 404 seem like a user error, so I don't think we need to log it at any level.
 - 403s are possibly a server error (if they're expecting it to be able to access a private repo but have the token misconfigured) but in larger deploys it's not something worth logging over, since users can enter arbitrary information.
 - 401s are probably a WARNING or ERROR, since that's something wrong with the token itself
 - 429s are a WARNING, since fundamentally it can be caused by users taking bad actions -- but something that affects the whole server so the admin should be more aware.

In none of these should we use `exception` since I don't think the stacktrace is useful.

Reply:
I have checked on rate limit status code is `403 rate limit exceeded` for Github api, this is the one we should sure show?

Reply:
Yes, I have showed warning on 401, 403 and 429.